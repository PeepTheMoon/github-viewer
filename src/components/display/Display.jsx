import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ name, username, followerCount, followingCount, profileLink }) => (
  <div>
    <p>Name: {name}</p>
    <p>Username: {username}</p>
    <p>Follower Count: {followerCount}</p>
    <p>Following Count: {followingCount}</p>
    <p>Profile link: {profileLink}</p>
  </div>
);

Display.propTypes = {
  name: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  followerCount: PropTypes.number.isRequired,
  followingCount: PropTypes.number.isRequired,
  profileLink: PropTypes.string.isRequired
};

export default Display;
