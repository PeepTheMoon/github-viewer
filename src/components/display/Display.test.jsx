import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Display name="Rachel Donahue" username="PeepTheMoon" followerCount={4} followingCount={5} profileLink="https://github.com/peepthemoon" />);
    expect(wrapper).toMatchSnapshot();
  });
});
