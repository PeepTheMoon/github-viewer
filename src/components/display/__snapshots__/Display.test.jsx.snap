// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders App 1`] = `
<div>
  <p>
    Name: 
    Rachel Donahue
  </p>
  <p>
    Username: 
    PeepTheMoon
  </p>
  <p>
    Follower Count: 
    4
  </p>
  <p>
    Following Count: 
    5
  </p>
  <p>
    Profile link: 
    https://github.com/peepthemoon
  </p>
</div>
`;
